### PR TITLE
add Ruby and Embedded Ruby to Languages

### PR DESCRIPTION
**Pull request by juice**

### CODE REVIEW

**Comment by juice** [comment]
merged newest code and fixed icons

File: styles/styles.css [hunk 1]
```diff
@@ -314,6 +314,16 @@
     content: "######";
 }
 
+/* JavaScript */
```
Comment (Hirse) [inline styles/styles.css]: Fix comment.

File: src/languages/Ruby.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    function getOutlineList(text, showArguments, showUnnamed) {
+        var lines = text.replace(/\)((?:[^\S\n]*\n)+)\s*\{/g, "){$1").split("\n");
+        var regexMethod = /\s*def\s+(\w*[.\w*]+\??)\s*(\([\w,\s,\$,\_="]*\))?/g;
+        var regexMethodClass = /\s*(class\s+<<\s+self)\s*/g;
+        var regexMethodPrivate = /\s*private\s*/g;
+        var result = [];
+        var isClassMethod = false;
+        var isPrivateMethod = false;
+        lines.forEach(function (line, index) {
+          var matchMethodClass = regexMethodClass.exec(line);
```
Comment (Hirse) [inline src/languages/Ruby.js]: Please use indentation of 4 spaces.

File: src/languages/Ruby.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+          if (matchMethod !== null) {
+            var name = matchMethod[1].trim();
+            var args = (matchMethod[2] || "");
+            var vis = "public";
+            if (isClassMethod || name.indexOf("self.") === 0) {
+              if (name.indexOf("self.") === 0) { name = name.slice(5); }
+              if (name[0] === "_") {
+                vis = "class-method-false-private"
+              } else {
+                vis = "class-method";
```
Comment (Hirse) [inline src/languages/Ruby.js]: Do you have a style for classes?

**Comment by juice** [comment]
thanks for pointing this out, if you find more let me know.

**Pull request merged**